### DISCUSSION TITLE
Custom Cookbooks / Collections (Add individual recipes)

### DISCUSSION THREAD

**Discussion by epheterson** [mealie-recipes/mealie#1435] [Feature Request]
**Before submitting this feature request I have**
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Cookbooks are a great way to find and share a collection of recipes, though it's only useful if the recipes are all of relatively similar types so that categories and tags can group them. If you want to create a custom collection, a user's favorites is the only way, and there can only be one. It'd be nice if there were a way to create custom collections of recipes to share.

**(Optional): Suggest A Solution**
When creating a cookbook, there could be two options – a filter/search style or an empty custom cookbook. I imagine each cookbook would only filter style or custom, but not both. If keeping the logic entirely separate is easier, the new type could just be called a Collection.

For custom cookbooks, you could click into it and use an "add recipe" button to search by name and add them, or on each recipe page there could be a new menu item "Add to cookbook".

Using the recipe page menu item, you could add to existing cookbooks, or create a new custom cookbook right there.

**Additional Information**
- [X] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by hay-kot** [comment]
I'm good with this. I think it makes sense to extend the cookbook functionality more so than create a new thing that'd we'd have to find space for on the frontend.

On the backend you'd need a way to connect recipes with the cookbooks via the SQL Database and on the frontend you'd need some way to select recipes to be added to a cookbook. Additionally, you'd need to update the Query we're using to grab all the recipes, which is probably going to be the hardest part.

**Comment by diedrichg** [comment]
This FR is in line with my request. I see cookbooks as a virtual version of a physical cookbook. I should be able to select a cookbook and see sections of the cookbook just as I would in real life - Appetizers, Side Dishes, Main Dishes, Breakfast, Lunch, Dinner, etc - or whatever Categories the user wishes to use in that cookbook. It's a hierarchy system - Cookbook>Categories>Recipes>Ingredients.

The cookbook should have the functionality to multi-select existing recipes or whole categories to add them to its contents.

The recipe should have a combobox with all existing Cookbooks with checkmark boxes next to them to add the recipe to selectively add that recipe to the desired cookbook(s). I guess there could also be a button to create a new cookbook on the fly.

As I see cookbooks now; all three of my cookbooks contain all my recipes and I do not see any way to remove or exclude recipes from their contents therefore having multiple cookbooks pointless.
